src/fetchers/tarball-fetcher.js

```diff
@@ -82,13 +82,30 @@ export default class TarballFetcher extends BaseFetcher {
         const expectHash = this.hash;
         const actualHash = validateStream.getHash();
         if (!expectHash || expectHash === actualHash) {
+          const tarballMirrorPath = this.getTarballMirrorPath();
+          const tarballCachePath = this.getTarballCachePath();
+
+          if (tarballMirrorPath) {
+            validateStream.pipe(fs.createWriteStream(tarballMirrorPath)).on('error', reject);
+          }
+
+          if (tarballCachePath) {
+            validateStream.pipe(fs.createWriteStream(tarballCachePath)).on('error', reject);
+          }
```

Comment:
Won't that cause the tarballs to still be written on the disk even if the validation fails?

Reply:
yup initial approach was `only write when hash matches` but realized there are other reasons that would cause failure. updated!